src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1492,8 +1492,13 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_WriteThreadNetw
             otOperationalDataset activeDataset;
             otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
             VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
-            uint64_t activeTimestamp = activeDataset.mPendingTimestamp;
-            err                      = encoder.Encode(activeTimestamp);
+#if OPENTHERAD_API_VERSION >= 219
+            uint64_t activeTimestamp = (activeDataset.mActiveTimestamp.mSeconds << 16) |
+                (activeDataset.mActiveTimestamp.mTicks << 1) | activeDataset.mActiveTimestamp.mAuthoritative;
+#else
+            uint64_t activeTimestamp = activeDataset.mActiveTimestamp;
+#endif
```

Comment:
Given that we have openthread as a submodule in the tree, do we need this backward compatibility?  I'm not finding other examples of #ifdef on the openthread API version in the tree.

Reply:
I was thinking the same, but it appears there are other pinned versions of OpenThread around. Specifically, I ran into this build failure on TELINK: https://github.com/project-chip/connectedhomeip/runs/7045847972?check_suite_focus=true

Reply:
Other examples of `OPENTHEAD_API_VERSION`:
```bash
$ git grep OPENTHREAD_API_VERSION
examples/common/pigweed/rpc_services/OtCli.h:static_assert(OPENTHREAD_API_VERSION >= 85, "Invalid Open Thread version");
examples/shell/shell_common/cmd_otcli.cpp:#if OPENTHREAD_API_VERSION >= 85
examples/shell/shell_common/cmd_otcli.cpp:#if OPENTHREAD_API_VERSION >= 85
examples/shell/shell_common/cmd_otcli.cpp:#if OPENTHREAD_API_VERSION >= 85
examples/shell/shell_common/cmd_otcli.cpp:#if OPENTHREAD_API_VERSION >= 85
examples/shell/shell_common/cmd_otcli.cpp:#endif // OPENTHREAD_API_VERSION >= 85
src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp:static_assert(OPENTHREAD_API_VERSION >= 156, "SRP Client requires a more recent OpenThread version");
src/platform/OpenThread/OpenThreadUtils.cpp:#if OPENTHREAD_API_VERSION >= 126
src/platform/OpenThread/OpenThreadUtils.cpp:#if OPENTHREAD_API_VERSION >= 126
```

Reply:
How is the API version mapped to Thread spec say 1.1 /1.2 /1.3 and so on

Reply:
OpenThread API version and Thread Protocol version are independent. The former tracks changes in OpenThread API, the latter tracks changes in Thread network protocol.

Reply:
Yeah... openthread comes from platforms in some cases and upgrading it all at the same time would require a lot more coordination (not to mention delay the fix waiting for the update to roll up the dependency tree).